src/feat/online-feature.h

```diff
@@ -117,13 +118,18 @@ class OnlineGenericBaseFeature: public OnlineBaseFeature {
   BaseFloat sampling_frequency_;
 
   // waveform_offset_ is the number of samples of waveform that we have
-  // already discarded, i.e. thatn were prior to 'waveform_remainder_'.
+  // already discarded, i.e. that were prior to 'waveform_remainder_'.
   int64 waveform_offset_;
 
   // waveform_remainder_ is a short piece of waveform that we may need to keep
   // after extracting all the whole frames we can (whatever length of feature
   // will be required for the next phase of computation).
   Vector<BaseFloat> waveform_remainder_;
+
+  // This object is used to down-sample the waveform, when
+  // sampling rate in frame extraction is less than
```

Comment:
sampling rate->the sampling rate; waveform sampling rate -> the waveform sampling rate.